### DISCUSSION TITLE
About Ice.BadMagicException

### DISCUSSION THREAD

**Discussion by Verminator86** [zeroc-ice/ice#2833] [Q&A] [answered]
I'm newbie with Icebox.
Our system currently use Icebox 3.4.2 on Windows environment. Could you please support how to solve Ice.BadMagicException?
Please refer to below image!

![BadMagicException](https://github.com/user-attachments/assets/c03d1aad-fb4c-4955-8e1a-d6184234ec7f)

**Comment by pepone** [comment, marked as answer]
This exception indicates that a message did not start with the expected magic number ('I', 'c', 'e', 'P'). The most likely cause for this is that you are connecting to non Ice server.

I would start by enabling network tracing by setting `Ice.Trace.Network=3` in both the client and server applications, and ensure that the client is connecting to the correct server.

**Reply by Verminator86** [reply]
Thank you for answering my question.
I changed configuration of Proxy then It's worked.
Before:  IceStorm.TopicManager.Endpoints=tcp -p 10000
After: IceStorm.TopicManager.Endpoints=tcp -p 10010
